optuna/integration/sklearn.py

```diff
@@ -367,7 +366,7 @@ def _partial_fit_and_score(
             if self.error_score == 'raise':
                 raise e
 
-            elif isinstance(self.error_score, Number):
+            elif isinstance(self.error_score, float):
```

Comment:
The current type hint for `self.error_score` is `Union[str, float]`, but the correspondence of `RandomSearchCV` in scikit-learn seems to be `str, Number` (c.f., https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/model_selection/_search.py#L1307-L1311).  So, I think keeping the `Number` here would avoid issues in real use cases.

Reply:
You are right. I relaxed the type hints to allow `numbers.Number`s.